Add colour scale to Anomaly map

rod-glover opened the pull request:
Resolves #78 
Resolves #89

rod-glover commented:
[Demo](https://services.pacificclimate.org/dev/weather-anomaly-viewer/)

@jameshiebert , a few things to note in this PR:
- Added colour scale to anomaly map, as requested in issue
- Added a clarifying annotation in other maps (they use single-colour markers)
- Added a "much greater than anything sane" range to the precip anomaly colour scale, coloured red. This clarifies what is going on (somewhat), but it also calls out our problem unmistakeably. Thoughts? Easy enough to revert or modify.

rod-glover commented on src/components/ColourScale/ColourScale.js:
```diff
@@ -0,0 +1,99 @@
+      </div>
+    );
+  }
+
```
The following section could use some comments.

rod-glover commented on src/components/DataMap/stationColor.js:
```diff
@@ -58,7 +57,7 @@ function value_to_color(value, thresholds, colors) {
 }
```
Suggest refactoring threshold and color values into single items to prevent possible errors in updating. E.g.,

```
const variableToPrecipColourScale = [
  { threshold: -87.5, color: '#8c510a' },
  ...
]
```

rod-glover commented:
[Demo](https://services.pacificclimate.org/dev/weather-anomaly-viewer/) redeployed

jameshiebert reviewed (commented):
Hi Rod,

The colour scale looks great and will help immensely on the usability. Two comments/critiques:

1. Did you get rid of the basemap fader? I actually find it very challenging to see the anomalies that lie in the middle/normal range because they blend in with the map so much. The basemap fader that is shown in the prod version really mitigates that problem.
2. I do like the fact that you've added a colour range for "implausibly high" values, though I agree with Lee that it would be helpful to separate that box on the colour scale (if easy to do). One thing that I struggle with is that using red *really* draws the user's attention to the errors and makes it hard to focus on the *good* data if that's your goal. That said, I can't think of a colour that one would want to use, that's outside of the range, but *doesn't* capture attention. I'm open to suggestions, experiments, or leaving it as is if we can't come up with something better.

rod-glover commented:
Thanks, James.

1. I did get rid of the basemap fader. I thought that the BC Albers Lite tiles were sufficiently muted, but evidently that is not the case for you, and I see your point. 
   - I can restore the fader. I'll do that in a separate PR. 
   - Alternatively, I can try using full-opacity markers. I'll try that first and if I like it I'll redeploy with that.
2. Done. Note altered colour, shape and annotation for the final interval; this is all driven by a config (when the config is externalized, even better). I changed the colour a little bit, but I don't think it improved things. I will continue to experiment.

[Demo](https://services.pacificclimate.org/dev/weather-anomaly-viewer/) redeployed

rod-glover commented:
[Demo](https://services.pacificclimate.org/dev/weather-anomaly-viewer/) redeployed

@jameshiebert 
1. I modified the markers, and I think they don't fade into the background the way they did before. It looks oddly pastel now, despite the colours not being changed, and the opacity being increased slightly to counteract that. But if that works for you, it works for me and it is simpler than adding a map fader.
2. Using a pale magenta, which is both distinct and less eye-catching (to my eye, at least). Thanks, @corviday , for the suggestion.

rod-glover commented:
I've addressed the feedback here and I think it is good enough for now. A PR coming soon will put these configurations into an external file so that future modifications are at deploy time.
